data-prepper-plugins/csv-processor/src/main/java/com/amazon/dataprepper/plugins/processor/csv/CsvProcessor.java

```diff
@@ -67,6 +74,7 @@ public Collection<Record<Event>> doExecute(final Collection<Record<Event>> recor
                     event.delete(config.getColumnNamesSourceKey());
                 }
             } catch (final IOException e) {
+                csvInvalidEventsCounter.increment();
```

Comment:
I'm not sure if I understood this, how is IOException related to `csvInvalidEventsCounter`?

Reply:
Jackson's `readValues`, `hasNextValue`, and `nextValue` methods throw a generic `IOException` when there's some issue with the input they're parsing. In testing, I could only get an IOException to throw if the csv record was malformed in some way such as forgetting to close a quote. So I figured that an IOException is a catch-all for invalid CSV Events. Do you think I should catch a more specific error? Perhaps a JSON Parse exception might make this function more readable.

Reply:
I think it's fine if Jackson throws IOException.